bot/extensions/adventofcode/commands.py

```diff
@@ -0,0 +1,216 @@
+
+    @app_commands.command()
+    async def subscribe(self, interaction: core.InteractionType) -> None:
+        """Subscribe to receive notifications for new puzzles"""
+
+        if self.role not in interaction.user.roles:
+            await interaction.user.add_roles(self.role)
+            await interaction.response.send_message(
+                "Okay! You have been __subscribed__ to notifications about new Advent of Code tasks."
+                "You can run `/aoc unsubscribe` to disable them again for you.",
+                ephemeral=True,
+            )
+        else:
+            await interaction.response.send_message(
+                "Hey, you already are receiving notifications about new Advent of Code tasks."
+                "If you don't want them any more, run `/aoc unsubscribe` instead.",
+                ephemeral=True,
+            )
+
+    @app_commands.command()
+    async def unsubscribe(self, interaction: core.InteractionType) -> None:
+        """Unsubscribe from receiving notifications for new puzzles"""
+
+        if self.role in interaction.user.roles:
+            await interaction.user.remove_roles(self.role)
+            await interaction.response.send_message(
+                "Okay! You have been __unsubscribed__ from notifications about new Advent of Code tasks.",
+                ephemeral=True,
+            )
+        else:
+            await interaction.response.send_message(
+                "Hey, you don't even get any notifications about new Advent of Code tasks currently anyway.",
+                ephemeral=True,
+            )
+
+    @app_commands.command()
+    async def countdown(self, interaction: core.InteractionType) -> None:
+        """Get the time left until the next puzzle is released"""
+
+        if (
+            int(datetime.now(tz=pytz.timezone("EST")).day) in range(1, 25)
+            and int(datetime.now(tz=pytz.timezone("EST")).month) == 12
+        ):
+            days = 24 - int(datetime.now().strftime("%d"))
+            hours = 23 - int(datetime.now().strftime("%H"))
+            minutes = 59 - int(datetime.now().strftime("%M"))
+
+            embed = discord.Embed(
+                title="Advent of Code",
+                description=f"There are {str(days)} days {str(hours)} hours "
+                f"and {str(minutes)} minutes left until AOC gets over.",
+            )
+            await interaction.response.send_message(embed=embed, ephemeral=True)
+
+        else:
+            await interaction.response.send_message("Advent of Code is not currently running.", ephemeral=True)
+
+    @app_commands.command(name="join")
+    async def join_leaderboard(self, interaction: core.InteractionType) -> None:
+        """Learn how to join the leaderboard"""
+
+        await interaction.response.send_message(
+            "Head over to https://adventofcode.com/leaderboard/private"
+            "with the code `975452-d90a48b0` to join the TWT private leaderboard!",
```

Comment:
Do we want to do the same with this code as I mentioned with the leaderboard id?